src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php

```diff
@@ -348,12 +348,20 @@ protected function describeContainerDefinition(Definition $definition, array $op
                     $argumentsInformation[] = sprintf('Service(%s)', (string) $argument);
                 } elseif ($argument instanceof IteratorArgument) {
                     $argumentsInformation[] = sprintf('Iterator (%d element(s))', \count($argument->getValues()));
+                    foreach (array_map(function (Reference $value) {return (string) $value; }, $argument->getValues()) as $service) {
+                        $argumentsInformation[] = sprintf('- %s', $service);
+                    }
                 } elseif ($argument instanceof ServiceLocatorArgument) {
                     $argumentsInformation[] = sprintf('Service locator (%d element(s))', \count($argument->getValues()));
                 } elseif ($argument instanceof Definition) {
                     $argumentsInformation[] = 'Inlined Service';
                 } else {
                     $argumentsInformation[] = \is_array($argument) ? sprintf('Array (%d element(s))', \count($argument)) : $argument;
+                    if (\is_array($argument)) {
+                        foreach (array_keys($argument) as $service) {
+                            $argumentsInformation[] = sprintf('- %s', $service);
```

Comment:
I reverted this part in 7a53e8d3f9 as I don't see how listing keys can be useful and why it's related to the PR description.